test/simulator/moves/imprison.js

```diff
@@ -0,0 +1,17 @@
+var assert = require('assert');
+var battle;
+
+describe('Imprison', function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
+	it('should prevent foes from using moves that the user knows', function () {
+		battle = BattleEngine.Battle.construct();
+		battle.join('p1', 'Guest 1', 1, [{species: 'Abra', ability: 'synchronize', moves: ['imprison', 'teleport']}]);
+		battle.join('p2', 'Guest 2', 1, [{species: 'Abra', ability: 'synchronize', moves: ['teleport', 'confusion']}]);
+		battle.commitDecisions();
+		battle.commitDecisions();
```

Comment:
We could save a turn and gain readibility if the first Abra had Prankster or the later had Lagging Tail.